transcom/mymove
# [MB-6578] UpdateWebhookSubscription endpoint

## Description

This is part of the Push Deploy work to add a new handler to update webhook subscriptions.

## Reviewer Notes

Please play close attention to the tests for UpdateWebhookSubscription Handler and Service Object

**The endpoint show allow updating of the following fields:**
subscriber_id - not the id of the subscription, but links to which contractor this subscription is related to status (ACTIVE, FAILING, DISABLED), event_key, callback_url, severity

## Setup
To test the endpoint in Postman
1. Add a new endpoint in postman for updateWebhookSubscription
2. Make sure that you are making a PATCH request with url: {{baseUrl}}/webhook_subscriptions/:webhookSubscriptionID
3. Construct a body similar to this example (this was generated by redoc)
```
{
  "subscriberId": "d494f114-05a2-4b39-840c-3d33243b7e29",
  "status": "ACTIVE",
  "callbackUrl": "string",
  "severity": 0,
  "eventKey": "string",
}
```

3. Follow [these instructions] (https://github.com/transcom/mymove/wiki/Test-Admin-and-Office-APIs-with-Postman) for copying the correct X-CSRF token into your Header in Postman
4. You should be able to send the request and receive a 200 response. Try some other variants to make sure they work.

To run tests

```sh
go test ./pkg/services/webhook_subscription -testify.m TestWebhookSubscriptionUpdater -failfast  -v
go test ./pkg/handlers/adminapi -testify.m TestWebhookSubscriptionUpdater -failfast  -v

```


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6578) for this change
